210 fix single msgs after batching

OsamaMSoliman opened the pull request:
### Why?
in `src/ml_board/backend/websocket_api/websocket_server.py` do we have `@self._socketio.on("mlgym_event")`
over the function `def on_mlgym_event(data):`  ??


(+ I'm still checking the batching process!)

OsamaMSoliman commented on src/ml_board/backend/websocket_api/websocket_server.py:
```diff
@@ -22,35 +22,119 @@ class WebSocketServer:
+        self._run = { "port": port, "host": host, "app": app }
+        # create the websocket
+        self._socketio = SocketIO(app, async_mode=async_mode, cors_allowed_origins=cors_allowed_origins, max_http_buffer_size=1e11)
+        # array of connected client session ids
```
why exactly is that?

le1nux commented on src/ml_board/backend/websocket_api/websocket_server.py:
```diff
@@ -8,7 +8,7 @@
 from ml_board.backend.websocket_api.checkpoint_cache import CheckpointCache, CheckpointEntity, CheckpointEntityTransferStatus
 from ml_gym.error_handling.exception import CheckpointEntityError
```
if they are not used --> delete :)

le1nux commented on src/ml_board/backend/websocket_api/websocket_server.py:
```diff
@@ -59,100 +143,26 @@ def _send_event_history_to_client(self, client_id: str, room_id: str):
-            print("Client disconnected", request.sid)
-            self._client_sids.remove(request.sid)
+####################################################################################################
+# Flask HTTP Server
+def create_flask_server() -> Flask:
+    # create Flask server
+    _app: Flask = Flask(__name__, template_folder="template")
+    _app.secret_key = 'secret!'
+    return _app
```
where was this previously implemented?

le1nux replied:
https://flask.palletsprojects.com/en/2.3.x/config/

OsamaMSoliman replied:
No where I took it from my implementation of the [dummy_server](https://github.com/mlgym/mlgym/blob/e3455fcfac62ed3477e6592bef4969908bc0ca19/src/ml_board/backend/dummy_server/dummy_server.py#LL14C1-L14C1)

based on [this](https://www.reddit.com/r/flask/comments/m0z7s1/need_some_help_understanding_the_use_of_a_flask/) and [this](https://stackoverflow.com/questions/34902378/where-do-i-get-secret-key-for-flask), I see setting the secret to a random string each time is not a bad idea for now, what do you think?

 `_app.secret_key = os.urandom(24)`

le1nux replied:
ok that sounds like a reasonable choice for now :)

le1nux reviewed (changes requested):
Left some comments :)

le1nux reviewed (changes requested):
Left some comments :)

OsamaMSoliman commented on src/ml_board/backend/websocket_api/websocket_server.py:
```diff
@@ -59,100 +143,26 @@ def _send_event_history_to_client(self, client_id: str, room_id: str):
+        port=5002,
+        async_mode=None,
+        app=create_flask_server(),
+        top_level_logging_path="/event_storage",
```
that's under `if __name__ == '__main__':` this file in the normal flow of the app isn't run like this! `ml_board_ws_endpoint` calls it, doesn't it?

le1nux replied:
oh true. I missed that :)

OsamaMSoliman commented on src/ml_board/backend/websocket_api/websocket_server.py:
```diff
@@ -22,35 +22,119 @@ class WebSocketServer:
-        self.app = app
-        self._socketio = SocketIO(app, async_mode=async_mode, cors_allowed_origins=cors_allowed_origins, max_http_buffer_size=100000000000)
+        # parameters for the run function
+        self._run = { "port": port, "host": host, "app": app }
```
I don't understand, can you give me an example?

I don't think you mean : `self._run_config = { "port": port, "host": host, "app": app }`, it's just an internal call, that made me think it's not so important how it looks as I'm not affecting how the function is being called from the outside?!

le1nux replied:
ah yeah it's fine. I thought you were passing them in like this `self.run(**self._run)`. 
Maybe renaming `_run` to `_run_params` would make sense though.

OsamaMSoliman commented:
@moinam please check the changes I made to `websocket_server.py`,
for example I deleted `self._checkpoint_cache = CheckpointCache()` because it private variable and it wasn't even being used anywhere else, but I could be wrong or maybe I changed this file while you were still working on it!

OsamaMSoliman commented on src/ml_board/backend/websocket_api/websocket_server.py:
```diff
@@ -27,35 +24,32 @@ class WebSocketServer:
 
     @property
-    def client_sids(self) -> List[str]:
+    def client_sids(self) -> list[str]:
```
I asked before about _"which python3 version are we using? 3.8 and earlier or 3.9 or 3.10+ ?"_ for this purpose 😅
